src/App.jsx

```diff
@@ -14,33 +16,92 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function getListOfGoods(goods, { sortField, checkReverse }) {
+  let listOfGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (sortField) {
+    listOfGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case 'alphabet':
+          return good1.localeCompare(good2);
+        case 'length':
+          return good1.length - good2.length;
+        
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (checkReverse) {
+    listOfGoods.reverse()
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  return listOfGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [checkReverse, setCheckReverse] = useState(false);
+  const visibleGoods = getListOfGoods(goodsFromServer, {
+    sortField,
+    checkReverse,
+  });
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => {
+            setSortField('alphabet');
+          }}
+          type="button"
+          className={`button is-info ${sortField === 'alphabet' ? '' : 'is-light'}`}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => {
+            setSortField('length');
+          }}
+          type="button"
+          className={`button is-success ${sortField === 'length' ? '' : 'is-light'}`}
+        >
+          Sort by length
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          onClick={() => {
+            checkReverse ? setCheckReverse(false) : setCheckReverse(true);
```

Comment:
Fix it 
<img width="806" alt="image" src="https://github.com/mate-academy/react_list-of-goods-js/assets/64750958/24a6eedf-781a-4686-a24b-683622f7541a">